pkg/asset/agent/manifests/nmstateconfig.go

```diff
@@ -73,61 +73,72 @@ func (n *NMStateConfig) Generate(dependencies asset.Parents) error {
+				if host.NetworkConfig.Raw != nil {
+					isNetWorkConfigAvailable = true
+
+					nmStateConfig := aiv1beta1.NMStateConfig{
+						TypeMeta: metav1.TypeMeta{
+							Kind:       "NMStateConfig",
+							APIVersion: "agent-install.openshift.io/v1beta1",
+						},
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      fmt.Sprintf(getNMStateConfigName(agentConfig)+"-%d", i),
+							Namespace: getNMStateConfigNamespace(agentConfig),
+							Labels:    getNMStateConfigLabelsFromAgentConfig(agentConfig),
+						},
+						Spec: aiv1beta1.NMStateConfigSpec{
+							NetConfig: aiv1beta1.NetConfig{
+								Raw: []byte(host.NetworkConfig.Raw),
+							},
+						},
+					}
+					for _, hostInterface := range host.Interfaces {
+						intrfc := aiv1beta1.Interface{
+							Name:       hostInterface.Name,
+							MacAddress: hostInterface.MacAddress,
+						}
+						nmStateConfig.Spec.Interfaces = append(nmStateConfig.Spec.Interfaces, &intrfc)
+
+					}
+					nmStateConfigs = append(nmStateConfigs, &nmStateConfig)
+
+					staticNetworkConfig = append(staticNetworkConfig, &models.HostStaticNetworkConfig{
+						MacInterfaceMap: buildMacInterfaceMap(nmStateConfig),
+						NetworkYaml:     string(nmStateConfig.Spec.NetConfig.Raw),
+					})
+
+					// Marshal the nmStateConfig one at a time
+					// and add a yaml seperator with new line
+					// so as not to marshal the nmStateConfigs
+					// as a yaml list in the generated nmstateconfig.yaml
+					nmStateConfigData, err := k8syaml.Marshal(nmStateConfig)
+
+					if err != nil {
+						return errors.Wrap(err, "failed to marshal agent installer NMStateConfig")
+					}
+					data = fmt.Sprint(data, fmt.Sprint(string(nmStateConfigData), "---\n"))
 				}
-				nmStateConfig.Spec.Interfaces = append(nmStateConfig.Spec.Interfaces, &intrfc)
-
 			}
-			nmStateConfigs = append(nmStateConfigs, &nmStateConfig)
 
-			staticNetworkConfig = append(staticNetworkConfig, &models.HostStaticNetworkConfig{
-				MacInterfaceMap: buildMacInterfaceMap(nmStateConfig),
-				NetworkYaml:     string(nmStateConfig.Spec.NetConfig.Raw),
-			})
+			if isNetWorkConfigAvailable {
+				n.Config = nmStateConfigs
+				n.StaticNetworkConfig = staticNetworkConfig
 
-			// Marshal the nmStateConfig one at a time
-			// and add a yaml seperator with new line
-			// so as not to marshal the nmStateConfigs
-			// as a yaml list in the generated nmstateconfig.yaml
-			nmStateConfigData, err := k8syaml.Marshal(nmStateConfig)
-
-			if err != nil {
-				return errors.Wrap(err, "failed to marshal agent installer NMStateConfig")
+				n.File = &asset.File{
+					Filename: nmStateConfigFilename,
+					Data:     []byte(data),
+				}
+				return n.finish()
 			}
-			data = fmt.Sprint(data, fmt.Sprint(string(nmStateConfigData), "---\n"))
-		}
-
-		n.Config = nmStateConfigs
-		n.StaticNetworkConfig = staticNetworkConfig
-
-		n.File = &asset.File{
-			Filename: nmStateConfigFilename,
-			Data:     []byte(data),
+		} else {
```

Comment:
Also to improve readability, would be possible to transform this block to match the happy path, ie something like:

```
...
   if len(agentConfig.Config.Hosts) == 0 {
     return nil
   }
   for i, host := range agentConfig.Config.Hosts {
     ...
   }```